taxi/views.py

```diff
@@ -87,3 +95,30 @@ class DriverListView(LoginRequiredMixin, generic.ListView):
 class DriverDetailView(LoginRequiredMixin, generic.DetailView):
     model = Driver
     queryset = Driver.objects.all().prefetch_related("cars__manufacturer")
+
+
+class AssignOrDeleteUserToCarView(LoginRequiredMixin, generic.View):
```

Comment:
You can put this logic into post method of CarDetailView. This way you don't need to create extra view and url endpoint to handle adding/deleting driver.
You will need to update action on  form on car detail template to send value to the cardetailview.